_source/_posts/2022-01-13-mobile-sso.md

```diff
@@ -0,0 +1,161 @@
+## `SFAuthenticationSession` or `ASWebAuthenticationSession` behavior
+The various browser options offer differing levels of cookie sharing in order to limit websites'  ability to track a user. 
+
+The cookie-sharing behavior is not well documented in Apple's documentation. The  [`SFSafariViewController` doc](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) mentions that, "In iOS 9 and 10, it shares cookies and other website data with Safari ... If you would like to share data between your app and Safari in iOS 11 and later, ... use `ASWebAuthenticationSession` instead."
+
+A few third-party websites extend Apple's documentation to highlight that persistent cookies are shared between an embedded browser and Safari. Unfortunately, it is rarely known that session cookies can also be shared between different embedded browsers. If your application just needs SSO between your mobile apps, you do not have to use a persistent cookie; a session cookie is sufficient. 
+
+The following table summarizes the complete sharing behavior in iOS 11 or later. I've omitted `SFAuthenticationSession` for brevity because it is deprecated, but it behaves the same as `ASWebAuthenticationSession`. I've also omitted the `prefersEphemeralWebBrowserSession` option for `ASWebAuthenticationSession` because when it is set, cookies are not shared at all. 
+ 
+<div markdown="1" class="scrollable">
+
+|   | APP1 + SFSafariViewController | App1 + ASWebAuthenticationSession | App2 + SFSafariViewController | App2 + ASWebAuthenticationSession | Safari | Other browsers (e.g., Chrome) |
+|-----|-----|-----|-----|-----|-----|-----|
+| Session cookie    |   | Yes |   |  Yes |     |   |
+| Persistent cookie |   | Yes |   |  Yes | Yes |   | 
+
+</div>
+
+In the following iOS cookie behavior demo video, you can see how the session cookie and the persistent cookie are shared. The session cookie sharing behavior is not intuitive. Even when App1 is closed (which should clear all session cookies), opening App 2 would make it possible to see App1's session cookie. 
+
+{% youtube mDytgEU0bO8 %}
+
+It is worth noting that in iOS 14 and later you can specify another browser, such as Chrome, as the default. This does not affect the sharing behavior. `SFSafariViewController` and `ASWebAuthenticationSession` always use Safari under the hood, so they will never share cookies with browsers other than Safari, even if they are set as the default. The term *system browser*, often used in online articles, is synonymous with Safari. 
+
+
+## Solution: how to remove the extra prompt
+
+Now that we've reviewed the evolution of iOS browser behavior , and explored the rationale for the changes, let's look at solutions to improve the user experience. 
+
+### Solution to problem 2
```

Comment:
This seems out of order,  "Solution to problem 2" comes before "Solution to problem 1"

Reply:
Solution to problem 1 will also solve problem 2 as a by product. The reverse sequence is to go with smaller scope / simpler solution first.

Reply:
AHH, I see! ❤️